app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/util/PublicKeyCredentialCreationOptionsExtensions.kt

```diff
@@ -0,0 +1,23 @@
+@file:OmitFromCoverage
+
+package com.x8bit.bitwarden.ui.vault.feature.util
+
+import com.x8bit.bitwarden.data.autofill.fido2.datasource.network.model.PublicKeyCredentialCreationOptions
+import com.x8bit.bitwarden.data.platform.annotation.OmitFromCoverage
+
+/**
+ * Returns true if the user verification should be prompted when registering this FIDO 2 credential.
+ */
+@Suppress("MaxLineLength")
+val PublicKeyCredentialCreationOptions.promptForUserVerification: Boolean
```

Comment:
⚠️ Maybe you're re-checking it in another place but `preferred` is only prompted if the device supports device based verification. So, if `preferred` but the device hasn't set any of the device based verification then you shouldn't block saving the item.

Reply:
We're only checking if the request itself indicates user verification should be prompted. If this evaluates to `true` we emit an event to the View ([here](https://github.com/bitwarden/android/pull/3375/files#diff-d47cca9817cd007369608a0acdc3ccd615fd91414e9be537d3fcfd0cebf47785R412) and [here](https://github.com/bitwarden/android/pull/3375/files#diff-68ecf581510809578c6ae3489a8519eda7c54ce870660dbe7002490d6dddd6eeR374)), which is then able to determine if device UV can be performed ([here](https://github.com/bitwarden/android/pull/3375/files#diff-07a8b79595539108ea8e4b3693f7973ec0d1ed9d2e058df0fc3dbd5e8d455093R127) and [here](https://github.com/bitwarden/android/pull/3375/files#diff-2a443b4356801c5960019b3a783534944eb74e79d8a1944d2d7d9478a4f8891eR149)). When device UV cannot be performed the View notifies the ViewModel by sending `BiometricsVerificationFailed` action ([here](https://github.com/bitwarden/android/pull/3375/files#diff-2a443b4356801c5960019b3a783534944eb74e79d8a1944d2d7d9478a4f8891eR149) and [here](https://github.com/bitwarden/android/pull/3375/files#diff-2a443b4356801c5960019b3a783534944eb74e79d8a1944d2d7d9478a4f8891eR161)).

Reply:
Ah. I see now. 👍